### PR TITLE
STYLE: Added from __future__ import annotations to required files

### LINKED ISSUE #41901: STYLE add check for future annotations
xref https://github.com/pandas-dev/pandas/pull/41892#issuecomment-857834651

### PR DESCRIPTION
**Pull request by aacosta13**
- [X] closes #41901 
- [ ] No tests added
- [X] All linting passed (used `isort` to add imports)
- What's New:

Built a [script](https://gist.github.com/aacosta13/eb1cb4b22ece354c60f0e7aab90137b0) using `isort` to add `from __future__ import annotations` to necessary files as referenced by issue #41901.

### CODE REVIEW

File: pandas/_version.py [hunk 1]
```diff
@@ -4,9 +4,10 @@
 # directories (produced by setup.py build) will contain a much shorter file
 # that just contains the computed version number.
 
+from __future__ import annotations
```
Comment (simonjayhawkins) [inline pandas/_version.py:7]: nit. can you move to adjacent to the other imports.

Comment (aacosta13) [reply pandas/_version.py]: Sure, moved it just now on top of the rest.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
there's some unrelated changes here, e.g. `pandas/tests/io/sas/data/DEMO_G.xpt` should be deleted

**Comment by aacosta13** [comment]
@MarcoGorelli sorry about that, could have sworn that file was marked as deleted before the PR. Just removed it, feel free to let me know if there is anything else to fix or take note of.

File: ci/print_skipped.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env python3
+from __future__ import annotations
```
Comment (jreback) [inline ci/print_skipped.py:2]: this shouldn't be added to non pandas/ files

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by MarcoGorelli** [comment]
closing as stale, feel free to let us know if you'd like to continue work on this

**Pull request closed without merging**